src/glog/logging.h.in

```diff
@@ -897,7 +897,7 @@ PLOG_IF(FATAL, GOOGLE_PREDICT_BRANCH_NOT_TAKEN((invocation) == -1))    \
 #define LOG_OCCURRENCES_MOD_N LOG_EVERY_N_VARNAME(occurrences_mod_n_, __LINE__)
 
 #define SOME_KIND_OF_LOG_EVERY_N(severity, n, what_to_do) \
-  static int LOG_OCCURRENCES = 0, LOG_OCCURRENCES_MOD_N = 0; \
+  thread_local int LOG_OCCURRENCES = 0, LOG_OCCURRENCES_MOD_N = 0; \
```

Comment:
This looks wrong to me. Imagine the same `LOG_EVERY_XXX()` called from several threads concurrently - every thread will get its own count now...

Reply:
I've made the counters `std::atomic<int>`, and now the only thing left is a fallback mechanism for pre C++11 compilers. Do you have any idea?

Reply:
Both GCC and MSVC provide atomic intrinsics. Please see https://gcc.gnu.org/onlinedocs/gcc/_005f_005fatomic-Builtins.html and https://docs.microsoft.com/en-us/windows/win32/api/winnt/nf-winnt-interlockedincrement.